spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/actions/SparkBinPackStrategy.java

```diff
@@ -60,12 +59,8 @@ public Set<DataFile> rewriteFiles(List<FileScanTask> filesToRewrite) {
       tableCache.add(groupID, table);
       manager.stageTasks(table, groupID, filesToRewrite);
 
-      // Disable Adaptive Query Execution as this may change the output partitioning of our write
```

Comment:
similar problems exist in `SparkSortStrategy` and `SparkZOrderStrategy`'s 
`rewriteFiles()` too.

So, it makes sense to add a method in `RewriteDataFilesSparkAction` which will return cloned spark session with disabled adaptive execution.

this new method can be called instead of `spark()` in these 3 places.
https://github.com/apache/iceberg/blob/master/spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/actions/RewriteDataFilesSparkAction.java#L491-L501

Reply:
Instead of the new method, `Override` is also fine as @kbendick suggested because `spark()` is called only once per `RewriteDataFilesSparkAction`

Reply:
> Instead of the new method, `Override` is also fine as @kbendick suggested because `spark()` is called only once per `RewriteDataFilesSparkAction`

+1